legal-api/src/legal_api/reports/report.py

```diff
@@ -148,29 +165,37 @@ def _get_environment():
         return ''
 
     def _get_template_filename(self, report_type=None):
-        if self._filing.filing_type == 'incorporationApplication' and report_type:
-            file_name = Report.incorporation_filing_reports[report_type]['fileName']
+        if report_type:
+            if report_type == 'noa':
+                file_name = 'noticeOfArticles'
+            else:
+                file_name = Report.additional_reports[self._filing.filing_type][report_type]['fileName']
             return '{}.html'.format(file_name)
         return '{}.html'.format(self._filing.filing_type)
 
     def _get_template_data(self, report_type=None):  # pylint: disable=too-many-branches
-        filing = copy.deepcopy(self._filing.filing_json['filing'])
-        if self._filing.filing_type == 'incorporationApplication':
-            self._format_incorporation_data(filing, report_type)
+        if report_type == 'noa':
+            filing = VersionedBusinessDetailsService.get_company_details_revision(self._filing.id, self._business.id)
+            self._format_noa_data(filing)
         else:
-            # set registered office address from either the COA filing or status quo data in AR filing
-            try:
-                self._set_addresses(filing)
-            except KeyError:
-                pass
+            filing = copy.deepcopy(self._filing.filing_json['filing'])
+            if self._filing.filing_type == 'incorporationApplication':
+                self._format_incorporation_data(filing, report_type)
 
-            # set director list from either the COD filing or status quo data in AR filing
-            try:
-                self._set_directors(filing)
-            except KeyError:
-                pass
+            else:
+                # set registered office address from either the COA filing or status quo data in AR filing
+                try:
```

Comment:
Just a style thing, but in python3 we  can suppress
eg.
from contextlib import suppress

with suppress(KeyError):
    self._set_addresses(filing)

Reply:
Done